src/Components/Admin/ObjectEditor/unorderedpuzzle.js

```diff
@@ -150,36 +144,37 @@ export default function UnorderedPuzzle(props) {
     };
 
     const addImage = () => {
+        if (!isUnordered) {
+            setImagesLen(images.length + 1);
+        }
         setImages([...images, {}]);
         props.addImage();
         setUploaded(false);
     };
 
     const deleteImage = (index) => {
-        const tempImages = JSON.parse(JSON.stringify(images));
+        const tempImages = images;
         tempImages.splice(index, 1);
+        props.saveImages(tempImages);
         setImages(tempImages);
-        props.deleteImage(index);
         setUploaded(false);
     };
```

Comment:
Can you explain this logic, intuitively the "deleteImage" function should be calling props.deleteImage haha, but I don't know the actual logic going on behind the scenes here.

Reply:
Yea, so the JSON.parse(JSON.stringify()) logic messes up the byte array attribute, so I switched it back to just = images, and the props.saveImages(tempImages) replaces the deleteImage line